docs/sphinx_source/guide/development-guide.md

```diff
@@ -0,0 +1,105 @@
+# `waveorder` development guide
+
+## Install `waveorder` for development
+
+1. Install [conda](https://github.com/conda-forge/miniforge) and create a virtual environment:
+
+    ```sh
+    conda create -y -n waveorder python=3.12
+    conda activate waveorder
+    ```
```

Comment:
Would it be useful to have a custom Conda `yaml` config for this package?

So a user can just do:

```shell
conda env create -f environment.yml
```